### PR TITLE
Adding functions in Python binder

### PR DESCRIPTION
**Pull request by Artix18**
Adding two functions:
- tc.shared_memory_size()
- TcExecutor.profile_kernel(input, (output))

### CODE REVIEW

File: tensor_comprehensions/__init__.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 from tensor_comprehensions.tclib import TcExecutor
 from tensor_comprehensions.tclib import Tuner
 from tensor_comprehensions.tclib import TunerConfig
+from tensor_comprehensions.tclib import shared_memory_size
```
Comment (ftynse) [inline tensor_comprehensions/__init__.py]: This is a quite low-level function, I would keep it in `tclib` and not import by default with `import tensor_comprehensions`.  We really want to keep the high-level API as simple as possible

File: tensor_comprehensions/pybinds/tclib.cc [hunk 2]
```diff
@@ -273,6 +273,23 @@ struct TcExecutor {
...
+
+  size_t profile(const py::tuple& inputs, const py::tuple& outputs) {
+    if (outputs.size() > 0) {
+      auto atOutputs = getATenTensors(outputs);
+      auto atInputs = getATenTensors(inputs);
+      tc::ProfilingInfo profinfo =
+          tc::aten::profile(*executor, atInputs, atOutputs);
+      return profinfo.kernelRuntime.toMicroSeconds();
+    } else {
+      auto atInputs = getATenTensors(inputs);
```
Comment (ftynse) [inline tensor_comprehensions/pybinds/tclib.cc]: Most of these lines are duplicates of the "if" block and can be hoisted.  You should be able to just use a ternary operator when assigning `atOutputs` and remove the branch completely.

File: tensor_comprehensions/pybinds/tclib.cc [hunk 3]
```diff
@@ -465,7 +482,13 @@ PYBIND11_MODULE(tclib, m) {
           "unchecked_run",
           &TcExecutor::uncheckedRun,
           py::arg("inputs"),
+          py::arg("outputs") = py::tuple())
+      .def(
+          "profile_kernel",
```
Comment (ftynse) [inline tensor_comprehensions/pybinds/tclib.cc]: Let's keep the names consistent with C++ API.  It should be called `profile`.

Comment (nicolasvasilache) [reply tensor_comprehensions/pybinds/tclib.cc]: profile returns a ProfileInfo structure that we don't want to expose, `profile_kernel` returns the kernel time in us.

**Pull request closed without merging**